src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -242,14 +244,22 @@ async def _load_scheduler(
         db_conn: SAConnection,
         sgroup_name: str,
     ) -> AbstractScheduler:
-        query = (
-            sa.select([scaling_groups.c.scheduler])
-            .select_from(scaling_groups)
-            .where(scaling_groups.c.name == sgroup_name)
-        )
-        result = await db_conn.execute(query)
-        scheduler_name = result.scalar()
-        return load_scheduler(scheduler_name, self.shared_config['plugins']['scheduler'])
+        try:
+            query = (
+                sa.select([scaling_groups.c.scheduler, scaling_groups.c.scheduler_opts])
+                .select_from(scaling_groups)
+                .where(scaling_groups.c.name == sgroup_name)
+            )
+            result = await db_conn.execute(query)
+            row = result.one()
+            scheduler_name = row['scheduler']
+            scheduler_opts = row['scheduler_opts']
+        except MultipleResultsFound as e:
+            log.exception('_load_scheduler(): multiple scheduler! \n{}', repr(e))
+        except NoResultFound as e:
+            log.exception('_load_scheduler(): scheduler does not exist! \n{}', repr(e))
+        scheduler_opts = {**self.shared_config['plugins']['scheduler'][scheduler_name], **scheduler_opts}
```

Comment:
What happens if the etcd does not include configs for the given scheduler at all? Does this handle that case as an empty dict?